src/components/snackbar/examples/snackbar-persistent-non-dismissible.tsx

```diff
@@ -0,0 +1,70 @@
+import { Component, Element, State, h } from '@stencil/core';
+
+/**
+ * Persistent and non-dismissible
+ * In some scenarios, a Snackbar that is persistent
+ * can simultaneously be non-dismissible.
+ *
+ * This is good for handling cases in which the system tries to
+ * communicate an ongoing background process or a status that the user
+ * cannot take an action on.
```

Comment:
Is snackbar a good component to use for this case? I guess it's less intrusive than a banner, but the banner is intended to be persistent and non-blocking to begin with. The snackbar is intended to be transient and non-blocking. I think it feels less like a jump to me, to make it possible to have a banner that cannot be dismissed by the user, than to have a snackbar that's persistent… 🤔

I'm a little worried over two things when it comes to using a snackbar for this:
1. the snackbar might cover things
2. you're only supposed to show one snackbar at a time (we don't have good functionality for solving that, but adding that shouldn't be too hard, and is something we ought to do), and since the snackbar is fairly frequently used, I think we run a rather significant risk of conflicts

Now, obviously, with a banner, we also run the risk of covering things (or did we solve that already?). But since the banner is _supposed_ to be persistent, having it move things out of the way (by adding some padding to the top of the page, for example), makes sense for its design regardless of whether it's dismissible by the user or not (imo). That's not the case for snackbar.

Reply:
> When you say "intended", I need to ask "intended by whom?" 😊

😄

> If you ask _me_, the banner (specially the current one, which is a ginormous block, covering and blocking everything below, specially the navigation passibility) is a horrible component, both when it comes to design and usability. Persistent banners would work perhaps a bit better if they would pop-up as a `sticky` component on the UI, so that they would still allow the user to scroll and interact with the rest of the page, and hopefully by doing so even be able to "fix" the issue/problem that the banner is warning about. But getting this behavior is so hard, specially from a component library's perspective. Because we never know how consumers will place the banner in the DOM and what scrolling structure or logic it will have. So the component itself cannot be so useful in this case. At least not in the CRM web client.

Yeah, makes sense. I agree, the banner shouldn't cover things, and it should _definitely_ be possible to scroll while it's showing. (But it's not preventing scrolling now, is it!? 😲)

> (we can't show several banners neither)

Yeah, absolutely true 👍

> When it comes to showing multiple snackbars, I wouldn't follow Material Design's old and silly conventions or rules here ("Don't show several snackbars!" or "snackbars shouldn't have icons" or "they should just have one action"). It took them 10 years to finally realize that _input fields cannot be just one line and a label on top of it_, even though they still don't acknowledge that and offer it as a valid style in their components.

I think there is at least one really good reason not to show several snackbars at once, and that's tied to the fact that they are (typically) only shown for a certain amount of time. If we show a snackbar for some given amount of time, let's say 10 seconds, you have 10 seconds to notice and read it. However, if we show two snackbars at the same time, each for 10 seconds, you effectively have only 5 seconds to read each.

So I don't think it's a silly convention that only one snackbar should be displayed at a time.

Reply:
> However, if we show two snackbars at the same time, each for 10 seconds, you effectively have only 5 seconds to read each.

Hm... Great argument. But there are many scenarios where information in the second snackbar could explain why the first snackbar was displayed. e.g.

_Snackabr 1:_

<table>
<tr><td>💀 Upload failed.    <kbd>Cancel</kbd> <kbd>Retry</kbd>
</table>



_Snackbar 2: (pending)_

<table>
<tr><td>🌍 You're currently offline…   <kbd>OK</kbd>
</table>

There are a ton of examples like this, where the user should take a decision, gathering information communicated from different sources. While your argument is true, I think the default time-out should be increased or programmatically controlled. 

Additionally, a healthy app needs a Notifications Center, in which all notifications will end up. Then users can always check again what the dismissed/missed info was about.

Reply:
Yeah, the cool thing about us having a service for all the different parts of the app to use to send notifications, is that we can absolutely implement these things nowadays. We could cue up notification after each other, or add time to one that's already being displayed, and store the same message somewhere so it can be viewed in a notification centre at a later time.

This is starting to sound like a pitch!

(Also, this stuff might be really useful to have sorted the day we want to enable server-side push, since we might want to notify the user that certain data updated in the background.)

Reply:
We actually got stuck on some limitation on our notification service in the web-client (or something like that). I don't really remember why we stopped working on this 😄